feed-integration-webapp/src/main/java/org/bahmni/module/feedintegration/atomfeed/client/AtomFeedProperties.java

```diff
@@ -0,0 +1,70 @@
+package org.bahmni.module.feedintegration.atomfeed.client;
+
+import java.io.InputStream;
+import java.util.Properties;
+
+public class AtomFeedProperties {
+
+
+    private static final String FEED_CONNECT_TIMEOUT = "feed.connectionTimeoutInMilliseconds";
+    private static final String FEED_REPLY_TIMEOUT = "feed.replyTimeoutInMilliseconds";
+    private static final String FEED_MAX_FAILED_EVENTS = "feed.maxFailedEvents";
+    private static final String FAILED_EVENT_MAX_RETRY = "feed.failedEventMaxRetry";
+
+    public static final String DEFAULT_PROPERTY_FILENAME = "/atomfeed.properties";
+
+    private Properties properties;
+
+    private static AtomFeedProperties atomFeedProperties;
+
+    private AtomFeedProperties() {
```

Comment:
We are loading from classpath application.properties .. is this a good idea? why not use a PropertyPlaceholderConfigurer or leverage spring externalized configuration support, so that anyone can override using either -D or using system variables or even using "--spring.config.location" at startup.

Reply:
I think you can just use @Value("${property-name}") and when a property source is configured .. it will automatically do the needful

Reply:
There were 2 such properties files, CraterProperties and AtomfeedProperties. CraterProperties was made to keep in line with how property files were being imported. I have changed it back to using @Value annotation. But unfortunately that can't be done to AtomfeedProperties as the whole class is being used in methods inside atomfeed dependency, which I can't edit. I also tried to use @Value to take a variable property name (just like how getProperty does) but @Value can't take variables thus doesn't work. So Properties related to Crater have been imported using @Value annotation in spring but AtomfeedProperties will have to stay the same unless atomfeed dependency is changed to not need that particular class.